### ISSUE TITLE
crash with segfault when starting the app

### ISSUE THREAD

**Issue by maximbaz** [wireapp/wire-android#1937]
## The problem

- Install Wire from Play store. 
- Launch the app, select "Login"
- Type anything in Email and Password fields, e.g. I'm typing `qqqqqq@qqqqqq.qqq` and `qqqqqq`.
- Press "arrow" button to login.

The app is instantly crashing at this point. If I launch it again, it will immediately crash again.

I checked that Wire has the following permissions enabled: Microphone, Storage and Camera.

This is a phone with x86 processor, so it might be contributing to the cause, however all other apps I tried work fine, only Wire cannot launch.

I connected the phone to the laptop and captured some logs, I can see there was a segfault:

```
4864:01-09 18:38:19.470   709  1798 I ActivityManager: Start proc 26269:com.wire/u0a147 for activity com.wire/com.waz.zclient.LaunchActivity
4870:01-09 18:38:19.500  2135  2147 D ForegroundUtils: Foreground changed, PID: 26269 UID: 10147 foreground: true
4873:01-09 18:38:19.500  2135  2147 D ForegroundUtils: UID: 10147 PID: 26269
4875:01-09 18:38:19.520 26269 26269 I MultiDex: VM with version 2.1.0 has multidex support
4876:01-09 18:38:19.520 26269 26269 I MultiDex: Installing application
4877:01-09 18:38:19.520 26269 26269 I MultiDex: VM has multidex support, MultiDex support library is disabled.
4878:01-09 18:38:19.520 26269 26269 I FirebaseInitProvider: FirebaseApp initialization unsuccessful
4889:01-09 18:38:19.730 26269 26269 V Monotype: SetAppTypeFace- try to flip, app = com.wire
4890:01-09 18:38:19.740 26269 26269 V Monotype:     Typeface getFontPathFlipFont - systemFont = /data/data/flipfont/app_fonts/Felbridge#Felbridge
4895:01-09 18:38:19.750 26269 26269 I art     : Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.app.JobIntentService$JobServiceEngineImpl>
4896:01-09 18:38:19.750 26269 26269 I art     : Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.app.JobIntentService$JobServiceEngineImpl>
4897:01-09 18:38:19.790 26269 26269 D FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
4898:01-09 18:38:19.790 26269 26269 D FirebaseApp: com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
4899:01-09 18:38:19.790 26269 26269 D FirebaseApp: com.google.android.gms.measurement.AppMeasurement is not linked. Skipping initialization.
4905:01-09 18:38:19.849 26269 26277 I KERNEL  : [40481.083329] binder: 26269:26277 refcount change on invalid ref 11
4911:01-09 18:38:19.970 26269 26269 V Monotype: SetAppTypeFace- try to flip, app = com.wire
4912:01-09 18:38:19.970 26269 26269 V Monotype:     Typeface getFontPathFlipFont - systemFont = /data/data/flipfont/app_fonts/Felbridge#Felbridge
4915:01-09 18:38:19.990 26269 26300 D OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
4935:01-09 18:38:20.030 26269 26300 I OpenGLRenderer: Initialized EGL, version 1.4
4936:01-09 18:38:20.030 26269 26300 W OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
4947:01-09 18:38:20.140 26269 26269 V Monotype: SetAppTypeFace- try to flip, app = com.wire
4948:01-09 18:38:20.140 26269 26269 V Monotype:     Typeface getFontPathFlipFont - systemFont = /data/data/flipfont/app_fonts/Felbridge#Felbridge
5036:01-09 18:38:21.410  2135  2830 D ForegroundUtils: UID: 10147 PID: 26269
5056:01-09 18:38:21.700 26269 26294 D FirebaseInstanceId: topic sync succeeded
6496:01-09 18:40:00.210  2135  2147 D ForegroundUtils: UID: 10147 PID: 26269
6622:01-09 18:40:02.540 26269 26269 I Choreographer: Skipped 44 frames!  The application may be doing too much work on its main thread.
6774:01-09 18:40:03.750  2135  2147 D ForegroundUtils: UID: 10147 PID: 26269
7692:01-09 18:40:21.280   304   304 F DEBUG   : pid: 26269, tid: 26433, name: pool-3-thread-6  >>> com.wire <<<
7729:01-09 18:40:21.513 26269 26433 I KERNEL  : [40602.710249] pool-3-thread-6[26433]: segfault at 73522eb6 ip 00000000de6d28ef sp 00000000da5680e8 error 7     <-----------------------------------
7730:01-09 18:40:21.513 26269 26433 I KERNEL  :  in base.odex[dda60000+108a000]
7731:01-09 18:40:21.513 26269 26433 I KERNEL  : 
7733:01-09 18:40:21.549   709 26622 I KERNEL  : [40602.746323] binder: 26269: binder_alloc_buf, no vma
7750:01-09 18:40:21.540   261   261 E lowmemorykiller: Error writing /proc/26269/oom_score_adj; errno=22
7754:01-09 18:40:21.540   709   728 W DisplayManagerService: Failed to notify process 26269 that displays changed, assuming it died.
7767:01-09 18:40:21.554   709   728 I KERNEL  : [40602.751866] binder: 26269: binder_alloc_buf, no vma
7777:01-09 18:40:21.560  2135  2830 D ForegroundUtils: UID: 10147 PID: 26269
7781:01-09 18:40:21.580   314   314 I Zygote  : Process 26269 exited due to signal (11)
7785:01-09 18:40:21.580   709   720 I ActivityManager: Process com.wire (pid 26269) has died
7786:01-09 18:40:21.580   709   720 D ActivityManager: cleanUpApplicationRecord -- 26269
7797:01-09 18:40:21.600  2135 18993 D ForegroundUtils: Process died; UID 10147 PID 26269
7798:01-09 18:40:21.600  2135 18993 D ForegroundUtils: Foreground changed, PID: 26269 UID: 10147 foreground: false
7922:01-09 18:40:23.350   709  1688 W InputMethodManagerService: Got RemoteException sending setActive(false) notification to pid 26269 uid 10147
```

Any ideas on what to try next are welcome.

## Environment

* Wire version that exhibits the issue: 3.24.684
* Last Wire version that did not exhibit the issue (if applicable): N/A
* Android OS version used to run Wire: 6.0.1
* Is this a custom firmware or a stock one: Stock
* Mobile phone model/manufacturer: Asus Zenfone 2 ZE551ML (ASUS_Z00AD)
* Mobile network type (EDGE/LTE/Wi-Fi/Offline): Wi-Fi, LTE

## Link to Debug logs

<details>
<summary>Collecting logs from Wire app</summary>

Create a [GIST](https://gist.github.com) which is a paste of your Wire logs, and link them here.
It is recommended to directly contact our [customer support](mailto:support@wire.com) and send the debug report by email, because it can potentially contain some private information, that you don't want to share to the public.

## How to send detailed debug report from Wire:

1. Go to Settings -> About
2. Tap "Wire swiss GmbH" 10 times
3. Reproduce your issue
4. Send the report from Settings -> Advanced -> Submit debug report
5. Go to Settings -> About
6. Tap "Wire swiss GmbH" 10 times again to disable detailed debug reports, because this slows down Wire

</details>

I'm unable to collect logs via Wire Settings as described above, because the menu is not present at the login step.

## Screenshots

N/A

**Comment by marcoconti83** [comment]
Hi @maximbaz ,
there are known issues with x86 processors. We had a look at possible causes but we could not find anything obvious in the code. This is a tricky issue and it will take time to track it down, as we don't even have a x86 device where we can reproduce this.

If anyone in the community would like to help on this, it would be amazing if someone with a x86 device could investigate by building the app in development mode and giving it a try on your device.

**Comment by marcoconti83** [comment]
Hi,
Better support for x86 has been released to public with version 3.42. I'll close the issue, feel free to reopen if this is still a problem.
